pkg/policy/api/v3/v2tov3.go

```diff
@@ -0,0 +1,537 @@
+// Copyright 2017-2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v3
+
+import (
+	"github.com/cilium/cilium/pkg/policy/api/v2"
+)
+
+// V2RulesTov3RulesSanitized translates the given v2Rules to the respective v3
+// Rules.
+// All given v2Rules needs to be sanitized and the returned v3 Rules will be
+// already sanitized. In case of an error while sanitizing a rule, that error
+// will be returned.
+func V2RulesTov3RulesSanitized(v2Rules *v2.Rules) (*Rules, error) {
+	v3Rules := V2RulesTov3Rules(v2Rules)
+	for _, v3Rule := range *v3Rules {
+		err := v3Rule.Sanitize()
+		if err != nil {
+			return nil, err
+		}
+	}
+	return v3Rules, nil
+}
+
+// V2RuleTov3RuleSanitized translates the given v2Rule to the respective v3
+// Rule.
+// The given v2Rules needs to be sanitized and the returned v3 Rule is alread
+// sanitized. In case of an error while sanitizing a rule, that error
+// will be returned.
+func V2RuleTov3RuleSanitized(v2Rule *v2.Rule) (*Rule, error) {
+	v3Rule := V2RuleTov3Rule(v2Rule)
+	if v3Rule == nil {
+		return nil, nil
+	}
+	err := v3Rule.Sanitize()
+	if err != nil {
+		return nil, err
+	}
+	return v3Rule, nil
+}
+
+// V2RulesTov3Rules translates the given v2Rules to the respective v3 Rules.
+// All given v2Rules needs to be sanitized and the returned v3 Rules needs to
+// be sanitized.
+func V2RulesTov3Rules(v2Rules *v2.Rules) *Rules {
+	if v2Rules == nil {
+		return nil
+	}
+	v3Rules := make(Rules, len(*v2Rules))
+	for i, v := range *v2Rules {
+		v3Rules[i] = V2RuleTov3Rule(v)
+	}
+	return &v3Rules
+}
+
+// V2RuleTov3Rule translates the given v2Rule to the respective v3 Rule.
+// The given v2Rule needs to be sanitized and the returned v3Rule needs to
+// be sanitized.
+func V2RuleTov3Rule(v2Rule *v2.Rule) *Rule {
+	if v2Rule == nil {
+		return nil
+	}
+	v3Rule := &Rule{}
+
+	v3Rule.EndpointSelector = *v2ESTov3ES(&v2Rule.EndpointSelector)
+
+	if v2Rule.Ingress != nil {
+		v3Rule.Ingress = []IngressRule{}
+
+		for _, v := range v2Rule.Ingress {
+			v3Rule.Ingress = append(v3Rule.Ingress, v2IRTov3IR(&v)...)
+		}
+	}
+
+	if v2Rule.Egress != nil {
+		v3Rule.Egress = []EgressRule{}
+
+		for _, v := range v2Rule.Egress {
+			v3Rule.Egress = append(v3Rule.Egress, v2ERTov3ER(&v)...)
+		}
+	}
+
+	v3Rule.Labels = v2Rule.Labels.DeepCopy()
+	v3Rule.Description = v2Rule.Description
+
+	return v3Rule
+}
+
+func v2ESTov3ES(v2ES *v2.EndpointSelector) *IdentitySelector {
```

Comment:
I'm not a big fan of these abbreviated function names. I prefer verbosity in this case (ES --> EndpointSelector).